src/main/java/org/elastos/did/DIDStore.java

```diff
@@ -362,41 +394,58 @@ public String publishDid(DID did, int confirms,
 		if (doc.isDeactivated())
 			throw new DIDStoreException("DID already deactivated.");
 
+		String lastTxid = null;
 		DIDDocument resolvedDoc = did.resolve();
 		if (resolvedDoc != null) {
 			if (resolvedDoc.isDeactivated())
 				throw new DIDStoreException("DID already deactivated.");
 
-			String localTxid = doc.getTransactionId();
-			if (localTxid == null)
-				throw new DIDStoreException("DID document not up-to-date");
+			if (!force) {
+				String localTxid = doc.getMeta().getTransactionId();
```

Comment:
can we write like below to get consistency with the way to acquire solvedTxId:
String localTxid = doc.getTransactionId()

Reply:
Maybe we need a public method (like getSignature()) to DIDDocument classs to get signature of this document.

Reply:
As internal implementation, here used the 'raw' mode to get the txid. I think it's ok.